### PR TITLE
Default include type signatures for all kind of builds

### PR DESCRIPTION
**Pull request by mrcarlberg**
This will make sure method and ivar type information is provided to the Objective-J runtime for all types of builds. It is done by providing the Objective-J compiler with option ```IncludeTypeSignatures```. It can optionally be turned off.

This will fix: #2398

A fix is also included that make sure that type signatures are copied to KVO method implementation for a object that is observed with KVO.

Some test cases are added for this.

### CODE REVIEW

File: Objective-J/CommonJS/lib/objective-j/compiler.js [hunk 1]
```diff
@@ -259,12 +259,16 @@ exports.main = function(args)
         if (argv[0] === "--help" || argv[0].substr(0, 1) == '-')
         {
             print("Usage (objjc 2.0): " + args[0] + " [options] [--] file...");
-            print("  -p, --print                    print the output directly to stdout");
-            print("  --unmarked                     don't tag the output with @STATIC header");
+            print("  -p, --print                         print the output directly to stdout");
+            print("  --unmarked                          don't tag the output with @STATIC header");
             print("");
-            print("  -T, --includeTypeSignatures    include type signatures in the compiled output");
+            print("  -T, --dont-include-type-signatures  include type signatures in the compiled output");
```
Comment (Dogild) [inline Objective-J/CommonJS/lib/objective-j/compiler.js]: Do we really want that ? What is the use case to not have type signatures ?

**Comment by Dogild** [comment]
Looks good to me ! I'll test it later in the day

File: Objective-J/CommonJS/lib/objective-j/compiler.js (hunk 1 above)
Comment (mrcarlberg) [inline Objective-J/CommonJS/lib/objective-j/compiler.js]: The use case is if you want to hide how your application works. Usually you minify/obfuscate the Javascript code, remove debug symbols and remove the type information. Sometimes your application need the type information and you have to leave it there.

**Comment by Dogild** [comment]
Tested here and it works !

**Comment by primalmotion** [comment]
Merged, Thanks!

**Comment by primalmotion** [comment]
+#fixed

**Pull request merged**